syntax = "proto3";

package pb;

import "UUID.proto";
//import "google/protobuf/timestamp.proto";

service UserController {
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
    rpc GetUser(GetUserRequest) returns (GetUserResponse) {}
    rpc ListUsers(ListUsersRequest) returns (ListUsersResponse) {}
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
}


message User {
    UUID id = 1;
    string password = 2;
    string username = 3;
    string email = 4;
}

message CreateUserRequest {
    User user = 1;
}

message CreateUserResponse {
    UUID id = 1;
}

message GetUserRequest {
    UUID id = 1;
}

message GetUserResponse {
    User user = 1;
}

message ListUsersRequest {
}

message ListUsersResponse {
    repeated User users = 1;
}

message UpdateUserRequest {
    User user = 1;
}

message UpdateUserResponse {
}

message DeleteUserRequest {
    UUID id = 1;
}

message DeleteUserResponse {
}